.github/workflows/ci.yaml

```diff
@@ -67,3 +68,49 @@ jobs:
 
       - name: Check format
         run: ./scripts/check_fmt.sh
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          # Needed to get older tags
+          fetch-depth: 0
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version: "~1.22"
+
+      - name: Login to GitHub Container Registry
+        if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      # do not push images for pull requests
+      - name: Build
+        if: github.event_name == 'pull_request'
+        run: |
+          VERSION=$(./scripts/version.sh)-dev-$(git rev-parse --short HEAD)
+          BASE=ghcr.io/coder/envbuilder-preview
+
+          ./scripts/build.sh \
+            --arch=amd64 \
+            --base=$BASE \
+            --tag=$VERSION
```

Comment:
I am not building a multi-arch here, as this step only ensures that Dockerfile builds fine.

Reply:
we should probably still build multi-arch as some changes may only cause issues with certain architectures

Reply:
Ok. Our build script uses `--load` when `--push` is not specified, which cannot be used for multi arch images. 
https://github.com/docker/buildx/issues/59

I didn't look deep if we actually needed to load the built images during local development otherwise I can just remove `--load` and build a multi arch image. 

While calling the script locally we only build with `$current` arch so `--load` works fine.